src/components/UnitContainer.js

```diff
@@ -19,12 +19,18 @@ export default class UnitContainer extends Component {
 
   update() {
     this.setState({
-      units: engine.boardState.getUnits().map(unit => <Unit progress={unit.progress} key={generateUUID()} />)
+      units: engine.boardState.getUnits().map(unit =>
+        <Unit
+          progress={unit.movementPattern.currentLocation}
+          color={unit.player.color}
+          key={generateUUID()}
+          />
+      )
     })
   }
 
   addUnit() {
-    engine.boardState.addUnit()
+    engine.spawnFriendlyUnit()
```

Comment:
I think the concept of "friendly" or "unfriendly" unit in terms of game engine is quite weird. Do we want to make a decision on having one, two, or three players, or human vs computer? How about game replays?